source/gloperate-glfw/source/Window.cpp

```diff
@@ -472,25 +471,25 @@ void Window::handleEvent(WindowEvent & event)
 
 void Window::clearEventQueue()
 {
-    while (!m_eventQueue.empty())
-    {
-        delete m_eventQueue.front();
-        m_eventQueue.pop();
-    }
+    std::queue<std::unique_ptr<WindowEvent>> empty;
+    std::swap(m_eventQueue, empty);
 }
 
 bool Window::createInternalWindow(const GLContextFormat & format, int width, int height, GLFWmonitor * /*monitor*/)
 {
     // Abort if window is already created
-    assert(nullptr == m_context);
+    assert(!m_context);
     if (m_context)
     {
         return false;
     }
 
     // Create GLFW window with OpenGL context
     GLContextFactory factory;
-    m_context = static_cast<GLContext*>(factory.createBestContext(format));
+    m_context = std::unique_ptr<GLContext>(
+                static_cast<GLContext*>(factory.createBestContext(format).release())
```

Comment:
Then I suggest adding the unique_ptr_cast functions to cppassist (including the deleter handling) and using them here.